src/black/__init__.py

```diff
@@ -500,6 +505,18 @@ def main(
     ctx.exit(report.return_code)
 
 
+def compute_exclude_and_gitignore(
+    exclude: Optional[Pattern[str]], root: Path
+) -> Tuple[Pattern[str], Optional["PathSpec"]]:
+    gitignore: Optional["PathSpec"]
+    if exclude is None:
+        exclude = re_compile_maybe_verbose(DEFAULT_EXCLUDES)
+        gitignore = get_gitignore(root)
+    else:
+        gitignore = None
+    return exclude, gitignore
```

Comment:
While I appreciate the idea of making small functions, I feel this is more indirection because the decision of computing the exclude pattern is moved to the computing function. And if the decision was moved back, this would be a two-line wrapper to two simple calls. But this is highly debatable. Thoughts?

This might be better if we get rid of the `exclude_and_gitignore` sentinel as I suggested in another comment, because we could unpack directly: `ex, ign = compute(ex, root)`.